Dockerfile.buf

```diff
@@ -1,4 +1,4 @@
-FROM --platform=${BUILDPLATFORM} golang:1.23-alpine3.21 as builder
+FROM --platform=${BUILDPLATFORM} golang:1.24-alpine3.21 as builder
```

Comment:
This upgrades go version.

Reply:
Only for our Docker image... which I think is fine?